tests/ui/traits/const-traits/const-traits-result.rs

```diff
@@ -0,0 +1,55 @@
+#![feature(const_trait_impl)]
```

Comment:
This can probably be a library test rather than UI. The error assertions that iterators aren't allowed don't seem overly useful here.

Reply:
`core` source files usually have a parallel test file in `coretests`, e.g. this one would be `library/coretests/tests/result.rs`